src/mnt_setup_fnc.cpp

```diff
@@ -8,59 +8,88 @@
 std::string const DATADIR = "/storage/epp2/phshgg/DVTuples__v23/";
 
 struct path_data {
-	std::string const DATADIR;
-	std::string sqrts,
+	std::string const DATADIR,
+	sqrts,
 	stripping,
 	year,
 	polarity,
 	stream;
 };
 
-void make_histogram(struct path_data path_in, std::string hist_name, std::string hist_setup, int const nbins, float const xmin, float const xmax, std::string expression, THStack *hist_stack) {
+struct input {
+	std::string const expression;
+	int const nbins;
+	float const xmin,
+	xmax;
+	std::string const label,
+	unit;
+};
+
+TH1F* make_TH1F(struct path_data path_in, struct input hist_input, std::string hist_name, std::string hist_setup) {
 	TChain ch("Z/DecayTree");
 	std::string path = path_in.DATADIR + path_in.sqrts + "TeV_" + path_in.year + "_" + path_in.stripping + "_" + path_in.polarity + "_" + path_in.stream + ".root";
 	ch.Add(path.c_str());
 	
-	TH1F *hist = new TH1F(hist_name.c_str(), hist_setup.c_str(), nbins, xmin, xmax);
-	expression += hist->GetName();
+	TH1F *hist = new TH1F(hist_name.c_str(), hist_setup.c_str(), hist_input.nbins, hist_input.xmin, hist_input.xmax);
+	std::string expression = hist_input.expression +">>"+ hist->GetName();
 	ch.Draw(expression.c_str());
-	hist_stack->Add(hist);
+	return hist;
 }
 
+void plot_histogram(struct path_data mnt_in, struct path_data sim_in, struct input hist_input) {
+	std::string hist_title = "Measurement vs Simulation for " + hist_input.label,
+		x_axis = hist_input.label + " " + hist_input.unit,
+		hist_text = hist_title + ";" + x_axis;
+	
+	TH1F *hist_mnt = make_TH1F(mnt_in, hist_input, "measurement", hist_text);
+	TH1F *hist_sim = make_TH1F(sim_in, hist_input, "simulation", hist_text);
+	
+	hist_sim->Scale(hist_mnt->Integral()/hist_sim->Integral());
+	
+	TCanvas canv;
+	hist_sim->SetStats(false);
+	hist_sim->Draw("HIST");
+	hist_mnt->Draw("SAME E");
+	std::string const filename = "Measurement_" + hist_input.label + ".png";
+	canv.SaveAs(filename.c_str());

```

Comment:
Can you try adding a legend with 
```cpp
canv.BuildLegend()
```
P.s. there is another way to add a legend with more control over the position, content, format etc... You would make a `TLegend` object. Where it is adequate I'd always favour the one line `BuildLegend()` option though.

Reply:
I've had a go at adding this to the code, but the title of the simulation data (here, "test title") is the same as the main title for the histogram (see attached). Is there a way to have these be different? 

I'd like to change the main title to be more descriptive of the whole histogram (e.g. "Measurement vs Simulation for Z_M"), whilst leaving the legend title as just "Simulation Data". Would this require using a `TLegend` object?
![Measurement_Z_M](https://user-images.githubusercontent.com/72559837/98260987-dd597d00-1f7b-11eb-85f6-6ee82d3b1181.png)

Reply:
Yes that is annoying. The workaround is to set both histogram titles to empty strings and change the name instead:
```
hist->SetTitle("");
hist->SetName("The name that you want to appear in the legend");
```

Reply:
P.s. it is conventional to draw the data (points with bars) in black and the model/simulation as a histogram with whatever colour you prefer. Though some colours e.g. `kAzure` in ROOT work better.